app/main.py

```
@@ -1,3 +1,65 @@
-class Distance:
-    # Write your code here
-    pass
+from __future__ import annotations
+
+
+class Distance:
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance | int | float) -> "Distance":
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        return Distance(self.km + other)
+
+    def __iadd__(self, other: Distance | int | float) -> "Distance":
+        if isinstance(other, Distance):
+            self.km += other.km
+            return self
+        self.km += other
+        return self
+
+    def __mul__(self, other: Distance | int | float) -> "Distance":
+        return Distance(self.km * other)
+
+    def __truediv__(self, other: Distance | int | float) -> Distance | None:

```

The type hints here need correction. The `other` parameter should only accept `int | float` according to the checklist. Consequently, the return type should just be `Distance`, not `Distance | None`, because the method should raise an exception for invalid types rather than returning `None`.